### ISSUE TITLE
Adding mean and median interpolation options in DEM generation process

### ISSUE THREAD

**Issue by Martin20494** [rosepearson/GeoFabrics#54]
Hi Rose,

It would be great if the DEM generation process can have the interpolation options among Inverse Distance Weighting (IDW), mean, median, and maybe some others. The reason is that if there is data problem, noise/ outliers, for example, the other options which will not be affected by this issue are still avaiable for usage.

Kind regards

Martin Nguyen

**Comment by rosepearson** [comment]
Thanks @Martin20494. I will hopefully address this alongside https://github.com/rosepearson/GeoFabrics/issues/65 soon. 

In this case, I plan to address this by adding a 'lidar_interpolation_type' attribute to the general section of the instruction file.

**Comment by rosepearson** [comment]
Hi @Martin20494 thanks for your input this morning. I've reviewed how that could be achieved within the netCDF framework, and I have decided I will initially introduce the ability to select which form of LiDAR interpolation (between mean, median and idw) where only one can be selected at a time. 

This is because this selection only applies to where LiDAR data dominates and for instance, not where bathymetry contours dominate. We can still look at moving to support the computation of multiple forms of LiDAR interpolation (i.e. idw and mean) in the future.
